File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/UserDefinedCells/readuserdefinedcells.rb
module Asposediagramjava
  module ReadUserDefinedCells
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "UserDefinedCells.vdx")

        shapes = diagram.getPages().getPage("Flow 1").getShapes()
        
        i = 0
        while i < shapes.getCount()
            shape = shapes.get(i)
            if shape.getNameU() == "Process"
                users = shape.getUsers()
                j = 0
                while j < users.getCount()
                    user = users.get(j)
                    puts user.getName() + ": " + user.getValue().getVal()
                    j +=1
                end
                break
            end
            i +=1
        end
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithUserdefinedCells/ReadUserDefinedCells.php
<?php
namespace Aspose\Diagram\WorkingwithUserdefinedCells;

use com\aspose\diagram\Diagram as Diagram;
class ReadUserDefinedCells{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram = new Diagram($dataDir . "drawing.vdx");

        $shapes = $diagram->getPages()->getPage("Flow 1")->getShapes();

        $i = 0;
        while($i<(int)(string)$shapes->getCount()) {
        $shape=$shapes->get($i);
        if($shape->getNameU()=="Process") {
        $users = $shape->getUsers();
        $j = 0;
        while ($j<(int)(string)$users->getCount()) {
            $user = $users->get($j);
            print $user->getName() . ": " . $user->getValue()->getVal();
            $j += 1;
        }
        break;
        }
        $i+=1;
        }
    }
}